User/Rajivts/UnitTestCasesForSystemXml

Rajivts opened the pull request:
Initial Test Case creation for System.Xml.XmlTextReader

krwq commented on src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderAttributeTests.cs:
```diff
@@ -0,0 +1,41 @@
+    public class XmlTextReaderAttributeTests
+    {
+        [Fact]
+        public void XmlTextReaderGetAttributeWithIndexTest()
```
I think this should be couple of more test cases:
- negative index
- 0 (as is), 1 (add 1 more attribute), 2 (out of bounds)

krwq replied:
also you can remove the prefix "XmlTextReader" from the test case name as it is already contained in the class name

krwq replied:
also one more test case where value is not there at all

krwq commented on src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderAttributeTests.cs:
```diff
@@ -0,0 +1,41 @@
+        }
+
+        [Fact]
+        public void XmlTextReaderMoveToAttributeWithNameTest()
```
please add another more test cases where: attribute is not there, there is more than one attribute with the same name (check if moved to the correct one)

krwq commented on src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderAttributeTests.cs:
```diff
@@ -0,0 +1,41 @@
+        }
+
+        [Fact]
+        public void XmlTextReaderMoveToAttributeWithNamespace()
```
- two namespaces and 1 attribute in each namespace - check if chosen the correct one
- why is the NameTable argument needed here?

krwq commented on src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderConstructionTests.cs:
```diff
@@ -0,0 +1,34 @@
+    public class XmlTextReaderConstructionTests
+    {
+        [Fact]
+        public void TextReaderConstructionWithStream()
```
I'd go with something like`CtorNullBaseUriStreamNameTable` or just `CtorNullBaseUri`

krwq commented on src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderPropertiesTests.cs:
```diff
@@ -0,0 +1,132 @@
+        [Fact]
+        public void XmlTextReaderHasValueTest()
+        {
+            XmlTextReader textReader = XmlTextReaderTestHelper.CreateReader("<element1 attr='val'> abc </element1>");
```
I'd unify some of those tests into one larger test cases with multiple asserts where possible

krwq commented on src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderTestHelper.cs:
```diff
@@ -0,0 +1,23 @@
+            return textReader;
+        }
+
+        internal static XmlTextReader CreateReader(string input, XmlNameTable nameTable)
```
I haven't seen where the nameTable was really used in the testcase. possibly we could remove it?

Rajivts commented:
@krwq Updated the PR with changes

Rajivts commented:
Removed the test based on BaseURI for XmlTextReader, since providing a non-null baseURI results in an exception preventing the tests from running. Also, fixed a couple more tests

Rajivts commented:
@krwq Made the relevant changes, PR is ready for review

krwq commented:
@Rajivts what error are you seeing for non-null baseURI? Is it perhaps a bug in XML?

krwq commented on src/System.Private.Xml/System.Private.Xml.sln:
```diff
@@ -158,136 +158,272 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "tests", "tests", "{1A2F9F4A
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "src", "src", "{E107E9C1-E893-4E87-987E-04EF0DCEAEFD}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "System.Xml.XmlTextReader.Tests", "tests\XmlTextReader\System.Xml.XmlTextReader.Tests\System.Xml.XmlTextReader.Tests.csproj", "{7FDF3054-AF3D-4804-88AE-37A9D62E4E3B}"
```
please put the project under `tests\Writers`

krwq replied:
yes 🤦‍♂️

krwq commented on src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderAttributeTests.cs:
```diff
@@ -9,33 +9,47 @@ namespace System.Xml.Tests
+            XmlTextReader textReader = XmlTextReaderTestHelper.CreateReader("<element1 attr='val' attr2='val2'> abc </element1>");
             Assert.True(textReader.Read());
-            Assert.True(textReader.MoveToAttribute("attr"));
+            Assert.True(textReader.MoveToAttribute("attr"));            
```
extra spaces in the end

krwq commented on src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderAttributeTests.cs:
```diff
@@ -9,33 +9,47 @@ namespace System.Xml.Tests
         {
-            XmlTextReader textReader = 
-                XmlTextReaderTestHelper.CreateReader("<List xmlns:tp='urn:NameSpace'><element1 tp:attr='val'>abc</element1></List>", new NameTable());
+            string inp = "<List xmlns:tp='urn:NameSpace'><element1 tp:attr='val' /><element1 xmlns:rp='urn' rp:attr='val2' /></List>";
```
`inp` -> `xml`

krwq commented on src/System.Private.Xml/tests/XmlTextReader/System.Xml.XmlTextReader.Tests/XmlTextReaderPropertiesTests.cs:
```diff
@@ -10,7 +10,7 @@ namespace System.Xml.Tests
     {
         [Fact]
-        public void XmlTextReaderHasValueTest()
+        public void HasValueTest()
```
please unify the test cases with the same input in this class

krwq replied:
(only for test cases where there is only ctor and property value check)

krwq commented:
Hey @Rajivts, do you think you will find some time to update this PR with the requested changes?

karelz commented:
@Rajivts ping?

karelz commented:
Closing for now. @Rajivts feel free to reopen once you have time. Thanks!
